### PR TITLE
Migrate Axiom dependency version to `1.4.0`

### LINKED ISSUE #37844: [Task]: Update the runtime axiom versions to latest
### Description

Current axiom version (1.2.2) have some graalvm. We need to bump to 1.3.0 or latest and check

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

Fixes #37844

## Approach
Modified the implementation according to the newer version.
https://ws.apache.org/axiom/release-notes/1.4.0.html

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: distribution/zip/jballerina-tools/LICENSE [hunk 1]
```diff
@@ -178,14 +178,14 @@ broker-coordination-0.970.0.jar
 ballerina-kafka-2.0.0-java.jar                                                                      jar            apache2
 org.wso2.carbon.messaging-2.3.7.jar                                                                 bundle         apache2
 ballerina-parser-2.0.0-Preview1.jar                                                                 jar            apache2
-axiom-api-1.2.22.jar                                                                                bundle         apache2
+axiom-api-1.4.0.jar                                                                                bundle         apache2
```
Comment (Nadeeshan96) [inline distribution/zip/jballerina-tools/LICENSE:181]:
```suggestion
axiom-api-1.4.0.jar                                                                                 bundle         apache2

```

File: distribution/zip/jballerina-tools/LICENSE [hunk 2]
```diff
@@ -178,14 +178,14 @@ broker-coordination-0.970.0.jar
...
 javax.transaction-api-1.2.jar                                                                       bundle         cddl1
 snakeyaml-1.32.0.wso2v1.jar                                                                         bundle         apache2
 ballerina-websub-1.0.0.jar                                                                          jar            apache2
 kotlin-stdlib-common-1.3.31.jar                                                                     jar            apache2
 netty-transport-4.1.39.Final.jar                                                                    bundle         apache2
 ballerina-stringutils-0.5.0.jar                                                                     jar            apache2
-axiom-impl-1.2.22.jar                                                                               bundle         apache2
+axiom-impl-1.4.0.jar                                                                               bundle         apache2
```
Comment (Nadeeshan96) [inline distribution/zip/jballerina-tools/LICENSE:188]:
```suggestion
axiom-impl-1.4.0.jar                                                                                bundle         apache2

```

File: distribution/zip/jballerina/LICENSE [hunk 3]
```diff
@@ -164,7 +164,7 @@ org.wso2.carbon.messaging-2.3.7.jar
 bir.jar                                                                                             jar            apache2
 xslt.jar                                                                                            jar            apache2
 observe.jar                                                                                         jar            apache2
-axiom-api-1.2.22.jar                                                                                bundle         apache2
+axiom-api-1.4.0.jar                                                                                bundle         apache2
```
Comment (Nadeeshan96) [inline distribution/zip/jballerina/LICENSE:167]:
```suggestion
axiom-api-1.4.0.jar                                                                                 bundle         apache2

```

File: distribution/zip/jballerina/LICENSE [hunk 4]
```diff
@@ -176,7 +176,7 @@ http.jar
 kotlin-stdlib-common-1.3.31.jar                                                                     jar            apache2
 netty-transport-4.1.39.Final.jar                                                                    bundle         apache2
 encoding.jar                                                                                        jar            apache2
-axiom-impl-1.2.22.jar                                                                               bundle         apache2
+axiom-impl-1.4.0.jar                                                                               bundle         apache2
```
Comment (Nadeeshan96) [inline distribution/zip/jballerina/LICENSE:179]:
```suggestion
axiom-impl-1.4.0.jar                                                                                bundle         apache2

```

File: distribution/zip/jballerina/LICENSE [hunk 5]
```diff
@@ -196,7 +196,7 @@ java.arrays.jar
 xml-1.1.0.jar                                                                                       jar            apache2
 lang.string.jar                                                                                     jar            apache2
 netty-codec-4.1.39.Final.jar                                                                        bundle         apache2
-axiom-dom-1.2.22.jar                                                                                bundle         apache2
+axiom-dom-1.4.0.jar                                                                                bundle         apache2
```
Comment (Nadeeshan96) [inline distribution/zip/jballerina/LICENSE:199]:
```suggestion
axiom-dom-1.4.0.jar                                                                                 bundle         apache2

```

File: distribution/zip/jballerina-tools/LICENSE (hunk 1 above)
Comment (Nadeeshan96) [inline distribution/zip/jballerina-tools/LICENSE:181]: Need to add a space here align the words. There are 4 more places like this below.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

**Review by gimantha — APPROVED** [review]

**Pull request merged**